bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/cache/ExpiringCacheAsync.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.cache;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Complementary class to {@link org.eclipse.smarthome.core.cache.ExpiringCache}, implementing an asynchronous variant
+ * of an expiring cache. An instance returns the cached value immediately to the callback if not expired yet, otherwise
+ * issue a fetch in another thread and notify callback implementors asynchronously.
+ *
+ * @author David Graeff - Initial contribution
+ *
+ * @param <V> the type of the cached value
+ */
+public class ExpiringCacheAsync<V> {
+    // The fields are package private on purpose, to allow easier unit testing.
+    final long expiry;
+    long expiresAt = 0;
+    CompletableFuture<V> currentNewValueRequest = null;
+    @Nullable
+    V value;
```

Comment:
IIRC this line break is added by the Eclipse IDE because of the missing visibility keyword.
IMHO a type annotation should be in front of the type and not a line above (same for interfaces where we omit the redundant public keyword on methods).

Perhaps the IDE settings will support this some time... So, not your fault.

Reply:
I was surprised as well about this format setting, but I didn't fight with Eclipse here and just accepted it. I might just declare the fields protected to correct this formatting.